src/Controller/UserController.php

```diff
@@ -4,33 +4,65 @@
 
 use App\Entity\User;
 use App\Repository\UserRepository;
+use App\Service\DatabaseService\BookDBService;
+use Doctrine\ORM\EntityManagerInterface;
 use Doctrine\ORM\Mapping\Id;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Security\Http\Attribute\CurrentUser;
 
 class UserController extends AbstractController
 {
-    public function __construct(
-        private UserRepository $userRepository
-    ) {}
-
-    public function displayFavorites(#[CurrentUser] ?User $user): Response
-    {
-        if ($user === null) {
-            return $this->json(['message' => 'missing or wrong credentials'], Response::HTTP_UNAUTHORIZED);
-        }
-        $user = $this->userRepository->find($user->getId());
-        return $this->json(['favorites' => $user->getFavorites()]);
+  public function __construct(
+    private UserRepository $userRepository,
+    private EntityManagerInterface $entityManager,
+    private BookDBService $bookDBService,
+  ) {
+  }
+
+  public function displayFavorites(#[CurrentUser] ?User $user): Response
+  {
+    if ($user === null) {
+      return $this->json(['message' => 'missing or wrong credentials'], Response::HTTP_UNAUTHORIZED);
+    }
+    $user = $this->userRepository->find($user->getId());
+    return $this->json(['favorites' => $user->getFavorites()]);
+  }
+
+  public function displayBorrowed(#[CurrentUser] ?User $user): Response
+  {
+    if ($user === null) {
+      return $this->json(['message' => 'missing or wrong credentials'], Response::HTTP_UNAUTHORIZED);
+    }
+
+    $user = $this->userRepository->find($user->getId());
+    return $this->json(['borrowed' => $user->getBorrowedBooks()]);
+  }
+
+  public function addFavoriteBook(#[CurrentUser] ?User $user, string $id): Response
+  {
+    $book = $this->bookDBService->searchBookById($id);
+
+    if ($book === null) {
+      return $this->json(['message' => 'Book not found'], Response::HTTP_NOT_FOUND);
+    }
+
+    if ($user === null) {
+      return $this->json(['message' => 'missing or wrong credentials'], Response::HTTP_UNAUTHORIZED);
```

Comment:
i would put the user check first (even before searching book), because if the user is not logged in, then this should not be possible to do